src/Raven.Server/Documents/Handlers/AI/Agents/AbstractSingleAiAgentProcessor.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client.Documents.Operations.AI.Agents;
+using Raven.Server.Documents.AI.AiGen;
+using Raven.Server.ServerWide.Context;
+using System.Net;
+using Raven.Client.Exceptions;
+using Raven.Server.Documents.Handlers.Processors.MultiGet;
+using Sparrow;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Sparrow.Server.Json.Sync;
+namespace Raven.Server.Documents.Handlers.AI.Agents
+{
+    internal abstract class AbstractSingleAiAgentProcessor : AbstractAiAgentProcessor<DatabaseRequestHandler, DocumentsOperationContext>
```

Comment:
this is not the name convention for a `non-sharded` database.
also, this is over abstraction at this point so let's remove it